libs/blocks/quiz/utils.js

```diff
@@ -152,41 +156,73 @@ const structuredFragments = (
   return structureFragments;
 };
 
-const nestedFragments = (
-  nestedFragsArray,
+export const nestedFragments = (
+  nestedFragsPrimaryArray,
+  nestedFragsSecondaryArray,
   resultResources,
   primaryProducts,
-  secondaryProductCodes,
+  secondaryProducts,
   isUmbrella,
 ) => {
   const nestedObject = {};
-  nestedFragsArray.forEach((frag) => {
+  nestedFragsPrimaryArray?.forEach((frag) => {
+    if (!frag) return;
     const fragKey = frag.trim();
-    const fragArray = [];
-    resultResources.data.forEach((row) => {
-      if (isUmbrella) {
-        // Get nested frags for all the primary products.
-        if (primaryProducts.length > 0 && primaryProducts.includes(row.product)) {
-          if (row[fragKey]) {
-            fragArray.push(row[fragKey]);
-          }
-        }
-      } else if (secondaryProductCodes.length > 0 && secondaryProductCodes.includes(row.product)) {
-        // Get nested frags for all the secondary products.
-        if (row[fragKey]) {
-          fragArray.push(row[fragKey]);
-        }
-      }
-    });
-    nestedObject[fragKey] = fragArray;
+    // eslint-disable-next-line max-len
+    nestedObject[fragKey] = getNestedFragments(resultResources, isUmbrella, primaryProducts, secondaryProducts, fragKey, true);
+  });
+  nestedFragsSecondaryArray?.forEach((frag) => {
+    if (!frag) return;
+    const fragKey = frag.trim();
+    // eslint-disable-next-line max-len
+    nestedObject[fragKey] = getNestedFragments(resultResources, isUmbrella, primaryProducts, secondaryProducts, fragKey, false);
   });
-
   return nestedObject;
 };
 
-const getRedirectUrl = (destinationPage, primaryProducts) => `${destinationPage}?primary=${primaryProducts}&quizKey=${quizKey}`;
+const getNestedFragments = (
+  resultResources,
+  isUmbrella,
+  primaryProducts,
+  secondaryProducts,
+  fragKey,
+  fetchForPrimaryProducts,
+) => {
+  const fragArray = [];
+  resultResources.data.forEach((row) => {
+    if (isUmbrella) {
+      // Get nested frags for all the primary products.
+      if (primaryProducts.length > 0 && primaryProducts.includes(row.product)) {
+        insertFragment();
+      }
+    } else if (isMatchingProductForFrag()) {
+      // Get nested frags for all the primary and secondary products.
+      insertFragment();
+    }
```

Comment:
In the end, you're calling `insertFragment` for each of the branches. You could go for something like
```
if ((isUmbrella && primaryProducts?.length && primaryProducts.includes(row.product))
  || isMatchingProductForFrag()) {
    insertFragment();
}
```
You might also want to check if `row.product` actually exists, in case both `primaryProducts` and `row.products` contain `undefined` entries